petastorm/spark/spark_dataset_converter.py

```diff
@@ -407,6 +410,41 @@ def _materialize_df(df, parent_cache_dir_url, parquet_row_group_size_bytes,
     return save_to_dir_url
 
 
+_DATABRICKS_WAIT_SECONDS_FOR_S3_CONSISTENCY = 30
+
+
+def _databricks_wait_for_s3_consistency(url_list):
+    """
+    This is used in databricks runtime. Waiting about 30 seconds to make sure
+    all files can be accessed by dbfs fuse.
+    This is because of s3 eventually consistency, and the backend of dbfs is s3.
+    """
+
+    if 'DATABRICKS_RUNTIME_VERSION' not in os.environ:
+        return
+
+    _, path_list = get_filesystem_and_path_or_paths(url_list)
+    remaining_list = list(path_list)
+    new_remaining_list = []
+
+    for _ in range(_DATABRICKS_WAIT_SECONDS_FOR_S3_CONSISTENCY):
+        for path in remaining_list:
```

Comment:
Let's add `logger.debug` to simplify debugging - in case a user wonders why the code hangs.